adam/learner/learner_utils.py

```diff
@@ -303,23 +303,27 @@ def in_left_to_right_order(semantic_nodes: Tuple[SemanticNodeWithSpan, ...]) ->
             previous_node = semantic_nodes[i]
         return True
 
-    def is_legal_template_span(candidate_verb_token_span: Span) -> bool:
+    def is_legal_template_span(
+        candidate_token_span: Span, *, invalid_token_spans: ImmutableSet[Span]
```

Comment:
Can you add a comment giving a sample user case for `invalid_token_spans` (that it, the one motivated by the current issue)